### PR TITLE
Cleanup path logic

### PR DESCRIPTION
**Pull request by xytovl**
Simplify logic for various path related functions, the goal is to reduce information duplication so it is easier to refactor installation layout.

### CODE REVIEW

File: alvr/filesystem-layout/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    // path to the manifest file for openVR
+    pub fn openvr_driver_manifest(&self) -> std::path::PathBuf {
+        self.openvr_driver_dir.join("driver.vrdrivermanifest")
+    }
+}
+
+lazy_static! {
+    pub static ref LAYOUT: Layout = if cfg!(windows) {
+        Layout {
+            openvr_driver_dir: std::path::PathBuf::from(""),
```
Comment (ronthecookie) [inline alvr/filesystem-layout/src/lib.rs:37]: Maybe use an [`Option`](https://doc.rust-lang.org/std/option/index.html)?

File: alvr/filesystem-layout/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            );
+        }
+        if !res.pop() {
+            return Err(
+                format! {"failed to get alvr dir from {}", component_path.to_string_lossy()},
+            );
+        }
+    }
+
+    println! {"alvr_dir_from_component: {} {} {}", component_path.to_string_lossy(), component_layout.to_string_lossy(), res.to_string_lossy()};
```
Comment (ronthecookie) [inline alvr/filesystem-layout/src/lib.rs:102]: Should probably be removed?

File: alvr/filesystem-layout/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+pub fn alvr_dir_from_component(
+    component_path: &std::path::PathBuf,
+    component_layout: &std::path::PathBuf,
+) -> Result<std::path::PathBuf, String> {
+    if component_layout.is_absolute() {
+        return Ok(std::path::PathBuf::from("/"));
+    }
+    let mut res = component_path.clone();
+    for component in component_layout.ancestors() {
+        if component == std::path::Path::new("") {
```
Comment (ronthecookie) [inline alvr/filesystem-layout/src/lib.rs:84]: Use a `use` for this and PathBuf. (rust-analyzer can do this automatically)

File: alvr/launcher/src/commands.rs [hunk 4]
```diff
@@ -214,5 +211,8 @@ pub fn invoke_installer() {
     spawn_no_window(Command::new(commands::installer_path()).arg("-q"));
 
     // delete crash_log.txt (take advantage of the occasion to do some routine cleaning)
-    fs::remove_file(current_alvr_dir().unwrap().join(CRASH_LOG_FNAME)).ok();
+    fs::remove_file(alvr_filesystem_layout::crash_log(
+        &current_alvr_dir().expect("get alvr dir"),
```
Comment (ronthecookie) [inline alvr/launcher/src/commands.rs:215]: Use [`trace_err` or `trace_none`](https://github.com/alvr-org/ALVR/wiki/How-ALVR-works#logging-and-error-management) please. No need to panic if we are returning a `StrResult`.

File: alvr/vulkan-layer/layer/layer.cpp [hunk 5]
```diff
@@ -125,18 +120,7 @@ VKAPI_ATTR VkLayerDeviceCreateInfo *get_chain_info(const VkDeviceCreateInfo *pCr
...
-
-    std::ifstream alvr_dir_if(path);
-    alvr_dir_if.getline(alvrDir, sizeof(alvrDir));
-
-    if (alvr_dir_if.fail()) {
-        std::cerr << "failed to read alvr_dir.txt\n";
-        exit(1);
-    };
-
+    // Make sure settings are loaded before we access it
```
Comment (ronthecookie) [inline alvr/vulkan-layer/layer/layer.cpp:123]: This is such a nitpick but: `s/it/them`. There are many settings.

**Review by ronthecookie — CHANGES_REQUESTED** [review]

File: alvr/filesystem-layout/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+            return Ok(res);
+        }
+        if res.file_name() != component.file_name() {
+            return Err(
```
Comment (ronthecookie) [inline alvr/filesystem-layout/src/lib.rs:88]: The early returns feel a bit unidiomatic, maybe see if you can somehow replace all these conditions with a `match` block?

File: alvr/filesystem-layout/src/lib.rs (hunk 1 above)
Comment (xytovl) [reply alvr/filesystem-layout/src/lib.rs:37]: Not for this one, on windows the path to the manifest is th same as alvr dir. Maybe "." would be more explicit?

Comment (ronthecookie) [reply alvr/filesystem-layout/src/lib.rs:37]: Oh yeah `.` sounds good.

File: alvr/launcher/src/commands.rs (hunk 4 above)
Comment (ronthecookie) [reply alvr/launcher/src/commands.rs:215]: It seems the web diff just messed up a bit and made me think this function was returning `StrResult` when, in fact, it isn't.

File: alvr/filesystem-layout/src/lib.rs (hunk 1 above)
Comment (xytovl) [reply alvr/filesystem-layout/src/lib.rs:37]: done, and added some comments on the struct

File: alvr/filesystem-layout/src/lib.rs (hunk 2 above)
Comment (xytovl) [reply alvr/filesystem-layout/src/lib.rs:102]: removed

File: alvr/filesystem-layout/src/lib.rs (hunk 3 above)
Comment (xytovl) [reply alvr/filesystem-layout/src/lib.rs:84]: done

File: alvr/filesystem-layout/src/lib.rs (hunk 6 above)
Comment (xytovl) [reply alvr/filesystem-layout/src/lib.rs]: I rewrote that part, is it more readable now?

Comment (ronthecookie) [reply alvr/filesystem-layout/src/lib.rs]: I can't really compare since you force-pushed but it seems good enough.

**Review by ronthecookie — APPROVED** [review]

**Pull request merged**